src/test-run/browser-manipulation-queue.js

```diff
@@ -1,75 +1,87 @@
-import { EventEmitter } from 'events';
 import shortId from 'shortid';
-import OS from 'os-family';
-import { resize as resizeWindow, getViewportSize } from 'testcafe-browser-natives';
+import { getViewportSize, generateThumbnail } from 'testcafe-browser-natives';
 import { isServiceCommand } from './commands/utils';
 import COMMAND_TYPE from './commands/type';
-import Warning from '../warnings';
+import WarningsLog from '../warnings/log';
 import WARNING_MESSAGE from '../warnings/message';
 
-export default class BrowserManipulationQueue extends EventEmitter {
-    constructor (windowId, screenshotCapturer) {
-        super();
 
+const SCREENSHOT_MANIPULATION_DESCRIPTION = 'take a screenshot';
+const RESIZE_MANIPULATION_DESCRIPTION     = 'do a window resize';
+
+export default class BrowserManipulationQueue {
+    constructor (browserConnection, screenshotCapturer) {
         this.commands           = [];
-        this.windowId           = windowId;
+        this.windowId           = browserConnection.id;
+        this.browserProvider    = browserConnection.provider;
         this.screenshotCapturer = screenshotCapturer;
+
+        this.warningLog = WarningsLog.get('manipulation:' + this.windowId);
```

Comment:
> What kind of warnings we can get from providers?
E.g. full functionality is not supported on this platform :); or that some optional dependency is missing; or some configuration was invalid, so provider used some default values, or if any non-critical error occurs. I think there might be a number of reasons to do not abort execution with an error, and just inform the user and simply retry or continue.

> Warnings are things that should go to reports, debug is used for internal debug messages, they shouldn't be mixed

Well, if somethings bad has happened, and `testcafe` hangs or just has aborted the execution, we will not be able to retrieve reports, and so in that case warnings will be lost. Also I can see in real-time if something is getting wrong if warning messages are piped to stdout.